SilKit/IntegrationTests/Hourglass/Test_HourglassNetSim.cpp

```diff
@@ -0,0 +1,1284 @@
+
+    if (cppControllerConfig.bufferConfigs.size() != cControllerConfig.numBufferConfigs)
+    {
+        return false;
+    }
+    for (size_t i = 0; i < cppControllerConfig.bufferConfigs.size(); i++)
+    {
+        if (!CompareFlexrayTxBufferConfig(cppControllerConfig.bufferConfigs[i], &cControllerConfig.bufferConfigs[i]))
+        {
+            return false;
+        }
+    }
+
+    if (cppControllerConfig.clusterParams.gColdstartAttempts != cControllerConfig.clusterParams->gColdstartAttempts
+        || cppControllerConfig.clusterParams.gCycleCountMax != cControllerConfig.clusterParams->gCycleCountMax
+        || cppControllerConfig.clusterParams.gdActionPointOffset != cControllerConfig.clusterParams->gdActionPointOffset
+        || cppControllerConfig.clusterParams.gdDynamicSlotIdlePhase
+               != cControllerConfig.clusterParams->gdDynamicSlotIdlePhase
+        || cppControllerConfig.clusterParams.gdMiniSlot != cControllerConfig.clusterParams->gdMiniSlot
+        || cppControllerConfig.clusterParams.gdMiniSlotActionPointOffset
+               != cControllerConfig.clusterParams->gdMiniSlotActionPointOffset
+        || cppControllerConfig.clusterParams.gdStaticSlot != cControllerConfig.clusterParams->gdStaticSlot
+        || cppControllerConfig.clusterParams.gdSymbolWindow != cControllerConfig.clusterParams->gdSymbolWindow
+        || cppControllerConfig.clusterParams.gdSymbolWindowActionPointOffset
+               != cControllerConfig.clusterParams->gdSymbolWindowActionPointOffset
+        || cppControllerConfig.clusterParams.gdTSSTransmitter != cControllerConfig.clusterParams->gdTSSTransmitter
+        || cppControllerConfig.clusterParams.gdWakeupTxActive != cControllerConfig.clusterParams->gdWakeupTxActive
+        || cppControllerConfig.clusterParams.gdWakeupTxIdle != cControllerConfig.clusterParams->gdWakeupTxIdle
+        || cppControllerConfig.clusterParams.gListenNoise != cControllerConfig.clusterParams->gListenNoise
+        || cppControllerConfig.clusterParams.gMacroPerCycle != cControllerConfig.clusterParams->gMacroPerCycle
+        || cppControllerConfig.clusterParams.gMaxWithoutClockCorrectionFatal
+               != cControllerConfig.clusterParams->gMaxWithoutClockCorrectionFatal
+        || cppControllerConfig.clusterParams.gMaxWithoutClockCorrectionPassive
+               != cControllerConfig.clusterParams->gMaxWithoutClockCorrectionPassive
+        || cppControllerConfig.clusterParams.gNumberOfMiniSlots != cControllerConfig.clusterParams->gNumberOfMiniSlots
+        || cppControllerConfig.clusterParams.gNumberOfStaticSlots
+               != cControllerConfig.clusterParams->gNumberOfStaticSlots
+        || cppControllerConfig.clusterParams.gPayloadLengthStatic
+               != cControllerConfig.clusterParams->gPayloadLengthStatic
+        || cppControllerConfig.clusterParams.gSyncFrameIDCountMax
+               != cControllerConfig.clusterParams->gSyncFrameIDCountMax)
+    {
+        return false;
```

Comment:
I checked the code coverage using gcov + lcov here, and the `return false` branches in the matchers are never taken, so I guess this implicityl tests the assignCToCxx and assignCxxToC code paths.